File: edu/src/main/java/io/agora/edu/common/service/HandsUpService.java
package io.agora.edu.common.service;

import io.agora.edu.common.bean.ResponseBody;
import io.agora.edu.common.bean.handsup.HandsUpReq;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface HandsUpService {
    @POST("edu/apps/{appId}/v2/rooms/{roomUUid}/processes/handsUp/progress")
    Call<ResponseBody<Integer>> applyHandsUp(
            @Path("appId") String appId,
            @Path("roomUUid") String roomUuid
    );

    @DELETE("edu/apps/{appId}/v2/rooms/{roomUUid}/processes/handsUp/progress")
    Call<ResponseBody<Integer>> cancelApplyHandsUp(
            @Path("appId") String appId,
            @Path("roomUUid") String roomUuid
    );

    @DELETE("edu/apps/{appId}/v2/rooms/{roomUUid}/processes/handsUp/acceptance")
    Call<ResponseBody<Integer>> exitHandsUp(
            @Path("appId") String appId,
            @Path("roomUUid") String roomUuid
    );
}

File: AgoraEduCore/src/main/java/io/agora/agoraeducore/core/internal/server/requests/http/retrofit/services/deprecated/HandsUpService.kt
package io.agora.agoraeducore.core.internal.server.requests.http.retrofit.services.deprecated

import io.agora.agoraeducore.core.internal.edu.common.bean.ResponseBody
import retrofit2.Call
import retrofit2.http.DELETE
import retrofit2.http.POST
import retrofit2.http.Path

interface HandsUpService {
    @POST("edu/apps/{appId}/v2/rooms/{roomUUid}/processes/handsUp/progress")
    fun applyHandsUp(
            @Path("appId") appId: String?,
            @Path("roomUUid") roomUuid: String?
    ): Call<ResponseBody<Int?>?>?

    @DELETE("edu/apps/{appId}/v2/rooms/{roomUUid}/processes/handsUp/progress")
    fun cancelApplyHandsUp(
            @Path("appId") appId: String?,
            @Path("roomUUid") roomUuid: String?
    ): Call<ResponseBody<Int?>?>?

    @DELETE("edu/apps/{appId}/v2/rooms/{roomUUid}/processes/handsUp/acceptance")
    fun exitHandsUp(
            @Path("appId") appId: String?,
            @Path("roomUUid") roomUuid: String?
    ): Call<ResponseBody<Int?>?>?
}